pkg/storage/fs/lister.go

```diff
@@ -24,7 +44,9 @@ func (l *storageImpl) List(ctx context.Context, module string) ([]string, error)
 	}
 	ret := []string{}
 	for _, fileInfo := range fileInfos {
-		if fileInfo.IsDir() {
+		if ok, err := l.isSemVerDir(module, fileInfo); err != nil {
```

Comment:
let's short-circuit this by removing the `isSemVerDir` and just doing this check: `fileInfo.IsDir() && len(fileInfo.Name()) > 2`

Reply:
@marpio pointed out that checking length > 2 wouldn't work because you might have `v10` which is len == 3 and so on and so forth. 

Basically, I'm hoping to get around making extra system calls because the /list endpoint is already a slow operation. 

Two ways to get around this is by 
1. Just checking if `!strings.Contains(version, ".")` 
2. By checking that the directory name is *not* `vN` where N is a valid integer

Reply:
@marwan-at-work @marpio thanks for suggestions about performance.

Only check `vN` is not enough, say we have a module `github.com/one/two` with some versions cached, a query for module `github.com/one` is valid which is expected to get nothing.

I committed another fix to only check semver without stats `go.mod` file, the only edge case left  uncovered i think is that a module contains a full semver part like `github.com/one/v1.1.1`, which maybe no need to deal with.

Reply:
@yittg then let's go with the first option, I think it should be enough because no major version will ever have a `.` in it. 
In other words, there shouldn't be a need to use `semver` package at all, right?

Reply:
I don't think so, dot in import path is valid besides in host part, maybe rarely though, not only in version. Full semver check is essential, i think.

Reply:
@yittg right, but when you're inside a module folder (in fs storage), there are no sub packages, there are only versions (v0.3.0/...) and major versions such as (v2/v2.1.3/...).

Reply:
would sub modules be included there by any chance?

Reply:
It might be worth making an example in github.com/athens-artifacts

Reply:
Say we have a repo `github.com/athens-artifacts/a.b.c`, athens will save a cache in fs by `go get github.com/athens-artifacts/a.b.c`, next `go get github.com/athens-artifacts` is a valid request may not correct will got a list contains `a.b.c` which is not a version though.

Reply:
@yittg that makes sense, I actually created https://github.com/athens-artifacts/submod-conflict where we can add real tests to (in another PR) 

You can notice that `v1.9.9` is actually not accepted as a sub-module. So option 2 actually does work (parsing vN). But sub-modules will also be included. So checking that semver.Canonical should account for both cases. 

Thanks again for looking into this :+1:

Reply:
yeah, some particular integrated tests for these senarios are required.